### DISCUSSION TITLE
How to create a tunel having ssh connection to server

### DISCUSSION THREAD

**Discussion by akushyn** [ronf/asyncssh#618] [Q&A]
I have a network setup , I can't assume the network structure.
I have ability to connect to 'root' server via ssh (username, password, port=2222).

server1 has connection to root
server2 has connection to server1
server3 has connection to server2

I can connect using port 22 inside network.
Want to use asyncssh package to walk through the network servers. 

```
        async with asyncssh.connect(
            "localhost",
            username='root',
            password='password',
            port=2222,
            known_hosts=None
        ) as conn:
            result = await conn.run("hostname")
            print(result.stdout)

```

this is how i create connection to root server.
please advise how to create tunnel to server 1, server2, server3 etc..

I wish i can get something similar to this, but cant get it how :

```
conn1 = conn.get_tunel(server1_ip, port=22)

# here i expect to print hostname of server1
result = conn1.run("hostname")
print(result.stdout)
```

```
root@root_host:~# ip route show | awk '/^[0-9]/ {print $1}'
10.10.100.0/24
```

```
root@root_host:~# nmap -p 22 --open 10.10.100.0/24 | awk '/Nmap scan report/{host=$NF; gsub(/[()]/, "", host)}/PORT/{print host, $NF}' | sed 's/SERVICE//'
10.10.100.2 
10.10.100.3 
10.10.100.1

```

**Comment by ronf** [comment]
If the "root" server supports port forwarding (or more specifically direct TCP/IP tunnels), you should be able to do something like:

```python
    async with asyncssh.connect('root_server', 2222) as conn:
        async with conn.connect_ssh('server1', 22) as conn1:
            result = await conn1.run('hostname')
            print(result.stdout)
```
